libavcodec/vvc/vvc_refs.c

```diff
@@ -138,6 +140,10 @@ static VVCFrame *alloc_frame(VVCContext *s, VVCFrameContext *fc)
         if (!frame->progress)
             goto fail;
 
+        ret = ff_hwaccel_frame_priv_alloc(s->avctx, &frame->hwaccel_picture_private);
```

Comment:
all software buffer allocation can be skipped, like rpl to progress
also all software delay frames can be removed. we do not need a delay for hw. we can output it immediately after vaEndPicture.

Reply:
Maybe better to keep software buffers there which can help check illegal syntax. And RPL may insert new ref frame to DPB in some special case, if skipped it, maybe cause unexpected error to HW. I will check how to remove defay frames.

Reply:
Delay logic in vvc native decoder may provide sync_depth function to HW like VAAPI encoder/vpp. And there is option that can turn OFF the delay. So I think it will be better to keep delay for HW too.